src/server/service/debug.rs

```diff
@@ -143,10 +147,55 @@ impl debugpb_grpc::Debug for Service {
 
     fn scan_mvcc(
         &self,
-        _: RpcContext,
-        _: ScanMvccRequest,
-        _: ServerStreamingSink<ScanMvccResponse>,
+        ctx: RpcContext,
+        req: ScanMvccRequest,
+        sink: ServerStreamingSink<ScanMvccResponse>,
     ) {
-        unimplemented!()
+        let debugger = self.debugger.clone();
+        let deal_future = self.pool.spawn_fn(move || {
+            let mut dealer = MvccKVToSink(Some(sink));
+            let result = match debugger.scan_mvcc(req, &mut dealer) {
+                Ok(_) => Ok(dealer.0.unwrap()),
+                Err(e) => Err((dealer.0.unwrap(), e)),
+            };
+            future::result(result)
+        });
+
+        let finish_future = deal_future.then(|v| match v {
+            Ok(mut sink) => {
+                let f = future::poll_fn(move || sink.close())
+                    .map_err(|e| Service::on_grpc_error("scan_mvcc", e));
+                Box::new(f) as Box<Future<Item = (), Error = ()> + Send>
+            }
+            Err((sink, e)) => {
+                let status = Service::error_to_status(e);
+                let f = sink.fail(status)
+                    .map_err(|e| Service::on_grpc_error("scan_mvcc", e));
+                Box::new(f) as Box<Future<Item = (), Error = ()> + Send>
+            }
+        });
+        ctx.spawn(finish_future);
+    }
+}
+
+struct MvccKVToSink(Option<ServerStreamingSink<ScanMvccResponse>>);
+
+impl MvccKVDealer for MvccKVToSink {
+    type Error = GrpcError;
+
+    fn deal(&mut self, key: Vec<u8>, mvcc: MvccInfo) -> Result<(), GrpcError> {
+        let mut mvcc_resp = ScanMvccResponse::new();
+        mvcc_resp.set_key(key);
+        mvcc_resp.set_info(mvcc);
+
+        let sink = self.0.take().unwrap();
+        match sink.send((mvcc_resp, WriteFlags::default())).wait() {
```

Comment:
Attach an `RpcContext` to `MvccKVToSink` and spawn the `sink.send()`, so that we dont need to call `wait()`.

Reply:
Because `sink.send` is called in other threads in `pool` and `RpcContext` can't `Send`, I think we can't do this.